L1Trigger/L1TTrackMatch/plugins/L1TkHTMissEmulatorProducer.cc

```diff
@@ -0,0 +1,286 @@
+
+/**\class L1TrackerHTMissEmulatorProducer L1TrackerHTMissEmulatorProducer.cc
+ L1Trigger/L1TTrackMatch/plugins/L1TrackerHTMissEmulatorProducer.cc
+ Description: Takes L1TTkJets and performs a integer emulation of Track-based missing HT, outputting a collection of EtSum 
+*/
+
+// Original Author:  Hardik Routray
+//         Created:  Mon, 11 Oct 2021
+
+// system include files
+#include <memory>
+#include <numeric>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "DataFormats/Math/interface/LorentzVector.h"
+#include "DataFormats/L1TCorrelator/interface/TkPrimaryVertex.h"
+#include "DataFormats/L1TCorrelator/interface/TkHTMiss.h"
+#include "DataFormats/L1TCorrelator/interface/TkHTMissFwd.h"
+#include "DataFormats/L1Trigger/interface/EtSum.h"
+#include "DataFormats/L1Trigger/interface/TkJetWord.h"
+#include "L1Trigger/L1TTrackMatch/interface/L1TkHTMissEmulatorProducer.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+using namespace l1t;
+
+class L1TkHTMissEmulatorProducer : public edm::stream::EDProducer<> {
+public:
+  explicit L1TkHTMissEmulatorProducer(const edm::ParameterSet&);
+  ~L1TkHTMissEmulatorProducer() override;
+
+private:
+  virtual void beginJob();
+  void produce(edm::Event&, const edm::EventSetup&) override;
+  virtual void endJob(); 
+
+  // ----------member data ---------------------------
+
+  bool debug_ = false;
+  bool saveLUTs_ = false;
+  bool displaced_;
+
+  // floats used for debugging
+  float jetMinPt__;            // [GeV]
+  float jetMaxEta__;           // [rad]
+  int minNtracksHighPt__;
+  int minNtracksLowPt__;
+
+  l1tmhtemu::pt_t jetMinPt_;
+  l1tmhtemu::eta_t jetMaxEta_;
+  l1tmhtemu::ntracks_t minNtracksHighPt_;
+  l1tmhtemu::ntracks_t minNtracksLowPt_;
+
+  std::vector<l1tmhtemu::phi_t> cosLUT_;
+  std::vector<l1tmhtemu::MHTphi_t> atanLUT_;
+  std::vector<l1tmhtemu::Et_t> magNormalisationLUT_;
+
+  std::string L1MHTCollectionName_;
+
+  const edm::EDGetTokenT<TkJetWordCollection> jetToken_;
+};
+
+L1TkHTMissEmulatorProducer::L1TkHTMissEmulatorProducer(const edm::ParameterSet& iConfig)
+  : jetToken_(consumes<TkJetWordCollection>(iConfig.getParameter<edm::InputTag>("L1TkJetEmulationInputTag"))) {
+  debug_ = (bool)iConfig.getParameter<bool>("debug");
+  saveLUTs_ = (bool)iConfig.getParameter<bool>("saveLUTs");
+  displaced_ = iConfig.getParameter<bool>("displaced");
+
+  // floats used for debugging
+  jetMinPt__ = (float)iConfig.getParameter<double>("jet_minPt");
```

Comment:
Same comment as for the `bool`. No need to cast to `float`.

Reply:
Also, why two `__`? Is there a convention I'm missing?

Reply:
casting issue fixed in the latest commit. 
the cuts in floats had __ in the suffix and were only used in debugging. They have been removed altogether in the latest commit to avoid confusion.